e2e-tests/src/lib.rs

```diff
@@ -1,24 +1,15 @@
 use codec::Compact;
-use log::info;
-use sp_core::sr25519;
-use sp_runtime::{generic, traits::BlakeTwo256, MultiAddress};
-use substrate_api_client::rpc::WsRpcClient;
-use substrate_api_client::{AccountId, Api, UncheckedExtrinsicV4, XtStatus};
+use sp_runtime::MultiAddress;
+use substrate_api_client::{AccountId, UncheckedExtrinsicV4};
 
 mod accounts;
 pub mod config;
 mod fee;
-pub mod rpc;
-pub mod session;
 mod staking;
 pub mod test;
 mod transfer;
 mod waiting;
 
-type BlockNumber = u32;
-type Header = generic::Header<BlockNumber, BlakeTwo256>;
```

Comment:
Much better, so we're using now the _real_ `Header`, `KeyPair` types from `runtime/src.lib.rs`? That should be good, as the types below are really sneaked from runtime anyway.

Reply:
Uhhh, I have no idea what you mean, I just moved these definitions into `common`. >.<"

Reply:
There's sometimes a challenge when writing e2e tests such that we need a concrete type from the runtime, but we don't want to add a runtime dependency to e2e Cargo.toml. In such case, one "reverse-engineers" runtime, and copy-paste that `type` to e2e tests, essentially increasing code duplication counted globally in `aleph-node`. Having one `type` in `common` accessible by the runtime and e2e decreases that duplication. 

A side note is there's an ongoing discussion whether we need to add `../bin/runtime` dependency to e2e-client Cargo.toml. but I think since we're moving some e2e code into `common`, then we've found a solution to this .
CC @fbielejec

Reply:
I see, I think this is out of scope for this change, but I understand why this would be useful.

I would be worried if we wanted to mainline `common` in its current shape, I thought of it more as of "common client things". I would argue that if we want to expose this kind of stuff we should make a separate "runtime types" crate and refer to that both in `runtime` and `common`. I was kind of under the impression that `primitives` was supposed to be this, but apparently not quite.

Anyway, I think this requires some designing to get right.